### PR TITLE
Add ADIOS Python Bindings

### PR DESCRIPTION
**Pull request by ax3l**
Follow-up to #5932, addion the numpy python bindings of ADIOS.

I could alternatively also put Python as a `host` dependency of the C library and always build both in the [`adios` recipe](https://github.com/conda-forge/adios-feedstock). We should be able to spare a python build matrix for the c library by separating the two.

### CODE REVIEW

**Comment by ax3l** [comment]
@scopatz this is a follow-up with a python binding of #5932

File: recipes/adios-python/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  url: http://users.nccs.gov/%7Epnorbert/{{ name }}-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  skip: True  # [win]
+
+requirements:
+  host:
+    - adios {{ version }}
```
[the remark this answers is not in the document]
Comment (scopatz) [reply recipes/adios-python/meta.yaml]: good to know!

File: recipes/adios-python/build.sh [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+# Python3 fixes
+if [[ "${PY_VER}" =~ 3 ]]
+then
+  find $SRC_DIR/utils -name "*.py" -exec 2to3 -w -n {} \;
+fi
+
+# numpy bindings
+cd wrappers/numpy
+make python
+python setup.py install
```
Comment (scopatz) [inline recipes/adios-python/build.sh]: You may want to use $PYTHON here to ensure that it uses the conda-build python, but this should be fine for now

File: recipes/adios-python/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: http://users.nccs.gov/%7Epnorbert/adios-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  skip: True  # [win]
+
+requirements:
+  host:
```
Comment (scopatz) [inline recipes/adios-python/meta.yaml]: Don't you need to add a build section with compilers? It seems like you are building C/C++ code through cython

File: recipes/adios-python/meta.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (ax3l) [reply recipes/adios-python/meta.yaml]: added now inline as comment as well :)

File: recipes/adios-python/build.sh [hunk 4]
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+if [ "$(uname)" == "Darwin" ]; then
+    # remove -lrt
+    sed -i '.bak' 's/ -lrt//g' $SRC_DIR/wrappers/numpy/Makefile
```
Comment (ax3l) [inline recipes/adios-python/build.sh]: Fix provided upstream in https://github.com/ornladios/ADIOS/pull/179 for future releases

**Pull request merged**